connectors/spark/src/test/java/io/unitycatalog/connectors/spark/SparkIntegrationTest.java

```diff
@@ -0,0 +1,209 @@
+        session.sql("INSERT INTO " + tableFullName + " SELECT 1, 'a'");
+        Row row = session.sql("SELECT * FROM " + tableFullName).collectAsList().get(0);
+        Assert.assertEquals(1, row.getInt(0));
+        Assert.assertEquals("a", row.getString(1));
+    }
+
+    private SchemaOperations schemaOperations;
+    private TableOperations tableOperations;
+
+    private void createCommonResources() throws ApiException {
+        // Common setup operations such as creating a catalog and schema
+        catalogOperations.createCatalog(CATALOG_NAME, "Common catalog for tables");
+        schemaOperations.createSchema(new CreateSchema().name(SCHEMA_NAME).catalogName(CATALOG_NAME));
+        catalogOperations.createCatalog(SPARK_CATALOG, "Spark catalog");
+        schemaOperations.createSchema(new CreateSchema().name(SCHEMA_NAME).catalogName(SPARK_CATALOG));
+    }
+    private void setupTables(
+            String catalogName,
+            String schemaName,
+            String tableName,
+            DataSourceFormat format,
+            SparkSession session) throws IOException, ApiException {
+        ColumnInfo c1 = new ColumnInfo().name("i").typeText("INTEGER")
+                .typeJson("{\"type\": \"integer\"}")
+                .typeName(ColumnTypeName.INT).typePrecision(10).typeScale(0).position(0)
+                .comment("Integer column")
+                .nullable(true);
+        ColumnInfo c2 = new ColumnInfo().name("s").typeText("STRING")
+                .typeJson("{\"type\": \"string\"}")
+                .typeName(ColumnTypeName.STRING).position(1)
+                .comment("String column")
+                .nullable(true);
+        String location = new File(new File(dataDir, catalogName), tableName).getCanonicalPath();
+        if (format == DataSourceFormat.DELTA) {
+            // The Delta path can't be empty, need to set up here.
+            session.sql(String.format("CREATE TABLE delta.`%s`(i INT, s STRING) USING delta", location));
+        }
+        CreateTable createTableRequest = new CreateTable()
+                .name(tableName)
+                .catalogName(catalogName)
+                .schemaName(schemaName)
+                .columns(Arrays.asList(c1, c2))
+                .comment(COMMENT)
+                .storageLocation(location)
+                .tableType(TableType.EXTERNAL)
+                .dataSourceFormat(format);
+        tableOperations.createTable(createTableRequest);
+    }
+
+    @Before
+    public void setUp() {
+        super.setUp();
+        schemaOperations = new SdkSchemaOperations(createApiClient(serverConfig));
+        tableOperations = new SdkTableOperations(createApiClient(serverConfig));
+        cleanUp();
+    }
+
+    @Override
+    protected CatalogOperations createCatalogOperations(ServerConfig serverConfig) {
+        return new SdkCatalogOperations(createApiClient(serverConfig));
+    }
+
+    @Override
+    protected void cleanUp() {
```

Comment:
Arent these functions already present in some form in the server tests?

Reply:
cc @ravivj-db ?

Reply:
Yes, but this test suite needs extra cleanup: the SPARK_CATALOG and the data folder for testing external tables.